src/main/java/de/srendi/advancedperipherals/common/util/LuaConverter.java

```diff
@@ -67,19 +99,12 @@ public static Map<String, Object> livingEntityToLua(LivingEntity entity, boolean
         return data;
     }
 
-    public static Map<String, Object> animalToLua(Animal animal, ItemStack itemInHand, boolean detailed) {
-        Map<String, Object> data = livingEntityToLua(animal, detailed);
-        data.put("baby", animal.isBaby());
-        data.put("inLove", animal.isInLove());
-        data.put("aggressive", animal.isAggressive());
-        if (animal instanceof IForgeShearable shareable && !itemInHand.isEmpty()) {
-            data.put("shareable", shareable.isShearable(itemInHand, animal.level, animal.blockPosition()));
-        }
-        return data;
-    }
-
     public static Map<String, Object> playerToLua(Player player, boolean detailed) {
         Map<String, Object> data = livingEntityToLua(player, detailed);
+        data.put("secondaryUseActive", player.isSecondaryUseActive());
+        data.put("hasContainerOpen", player.hasContainerOpen());
```

Comment:
what's the usage of `hasContainerOpen` here?

Reply:
It's true if the player has a container open that isn't their inventory. Could be used to detect players opening non-trapped chests. In general I have made as much info available as possible as players will probably find a use for all of it.

Reply:
imo if you provide `hasContainerOpen` here you should also provide which container the player opened (e.g. give a block position of the inventory, and ignore when player opened any container which is not a block container), because the player can open a container in a massive environment and the value will be kinda useless

Reply:
I can't think of a reliable way to get the block position of the container the player opened. The only thing I can think of is using `Entity.pick` to cast a ray when `playerToLua` is called. The issue is `playerToLua` could be called at any time, and a mob may have walked in front of the player while the player has the container open, which means the block position wouldn't be found.

Reply:
Alr, maybe just give the `containerId` as well as `stateId` then

Reply:
Those aren't unique unfortunately. After talking on the NeoForge discord this can't really be done without large caveats, which would be difficult to explain to the player. Still, I think `hasContainerOpen` could still be used to for example make a sneaky turtle that only moves when the player is distracted looking in blocks. It could also be used in combination with getting the players position to determine what block they are interacting with in a restricted environment like a minigame. I've decided to make `hasContainerOpen` part of the `detailed` info.